src/Electrical/Digital/logic_vectors.jl

```diff
@@ -0,0 +1,47 @@
+import Base: size, axes, getindex
+
+const LogicOrNumber = Union{Logic, Number}
+
+struct StdULogicVector{N} <: AbstractArray{Logic, N}
+    logic::AbstractArray{Logic}
+    level::AbstractArray{Int}
+    function StdULogicVector(l::AbstractArray)
+        N = ndims(l)
+        l = AbstractArray{Logic}(convert.(Logic, l))
+        new{N}(l, get_logic_level.(l))
+    end
+end
+
+struct StdLogicVector{N} <: AbstractArray{Logic, N}
+    logic::AbstractArray{Logic}
+    level::AbstractArray{Int}
+    function StdLogicVector(l::AbstractArray)
+        N = ndims(l)
+        l = AbstractArray{Logic}(convert.(Logic, l))
+        new{N}(l, get_logic_level.(l))
+    end
+end
+
+const LogicVector = Union{StdULogicVector, StdLogicVector}
+
+Base.size(l::L) where {L <: LogicVector} = size(l.logic)
+
+Base.axes(l::L) where {L <: LogicVector} = axes(l.logic)
```

Comment:
```suggestion
Base.axes(l::LogicVector) = axes(l.logic)
```
Same here, the type parameter should really only be introduced if you need to use it (or if you need to force specialization of a function type in some special cases).